esx_service/vmdk_ops.py

```diff
@@ -739,16 +744,34 @@ def get_full_vol_name(vmdk_name, datastore):
 
 
 # TBD - move to vmdk_utils
-def get_datastore_name(config_path):
-    """Returns datastore NAME in config_path (not url-name which may be used in path)"""
-    # path is always /vmfs/volumes/<datastore>/... , so extract datastore:
-    config_ds_name = config_path.split("/")[3]
-    ds_name = [x[0] for x in vmdk_utils.get_datastores() \
-                    if x[0] == config_ds_name or x[1] == config_ds_name ]
-    if len(ds_name) != 1:
-        logging.error("get_datastore_name: found more than one match: %s" % ds_name)
-    logging.debug("get_datastore_name: path=%s name=%s" % (config_ds_name, ds_name))
-    return ds_name[0]
+def get_datastore_url_from_config_path(config_path):
+    """Returns datastore url in config_path """
+    # path can be /vmfs/volumes/<datastore_url_name>/...
+    # or /vmfs/volumes/datastore_name/...
+    # so extract datastore_url_name:
+    config_ds_url = os.path.join("/vmfs/volumes/", os.path.realpath(config_path).split("/")[3])
```

Comment:
I personally think "/vmfs/volumes" is more readable in the context of the code, so will drop this comment.